build/Targets/Imports.targets

```diff
@@ -520,4 +520,30 @@
     </ItemGroup>
   </Target>
 
+  <!-- 
+    Prepare inputs for Symbol Store publishing.
+     
+    The imported targets populate ArtifactsSymStoreDirectory with links to binaries and Windows PDBs produced either 
+    directly by the build or converted from Portable PDBs produced by the build.
+    In official build the content of ArtifactsSymStoreDirectory is uploaded to a symbol server.
+  -->
+  <PropertyGroup>
+    <UsingToolPdbConverter>true</UsingToolPdbConverter>
+    <PublishOutputToSymStore Condition="'$(Nonshipping)' == 'true'">false</PublishOutputToSymStore>
+    <ArtifactsSymStoreDirectory Condition="'$(DeveloperBuild)' != 'true' and '$(OS)' == 'Windows_NT'">$(BaseOutputPath)$(Configuration)\SymStore\</ArtifactsSymStoreDirectory>
+  </PropertyGroup>
+
+  <Import Project="RepoToolset\SymStore.targets" />
+
+  <Import Project="RepoToolset\SourceLink.targets" Condition="'$(EnableSourceLink)' == 'true' and '$(GitHeadSha)' != ''"/>
+
+  <PropertyGroup>
+    <PathMap Condition="'$(DeterministicSourceRoot)' != ''">$(RepoRoot)=$(DeterministicSourceRoot)</PathMap>
+  </PropertyGroup>
+
+  <Target Name="_CheckForSpaces"
+          BeforeTargets="CoreCompile">
+    <Error Text="PathMap, DeterministicSourceRoot and RepoRoot shall not contain spaces (https://github.com/dotnet/roslyn/issues/22835)" 
+           Condition="$(PathMap.Contains(' '))" />
```

Comment:
We'd like to not make this worse. Do we have a middle ground we can do somehow?

Reply:
This shouldn't fail on a dev machine, unless they are testing CI build. PathMap is only set when DeterministicSourceRoot is set and that's only on build machines.

---
In reply to: [147292750](https://github.com/dotnet/roslyn/pull/22850#discussion_r147292750) [](ancestors = 147292750)